apm-agent-core/src/main/java/co/elastic/apm/agent/util/PackageScanner.java

```diff
@@ -87,13 +87,13 @@
         Files.walkFileTree(basePath, new SimpleFileVisitor<Path>() {
             @Override
             public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
-                if (file.toString().endsWith("class")) {
+                if (file.toString().endsWith(".class") || file.toString().endsWith(".esclazz")) {
                     // We need to escape both the filesystem-specific separator and the explicit `/` separator that may be added by the relativize() implementation
                     String classNameSuffix = basePath.relativize(file).toString()
                         .replace(System.getProperty("file.separator"), ".")
                         .replace('/', '.')
                         .replace(".class", "")
-                        .replace(".esclass", "");
+                        .replace(".esclazz", "");
```

Comment:
[question] should the package scanning that we use here transparently find class names that are re-packaged within agent classpath within `agent/` and `cached-lookup-key/` subfolders ? We don't remove any prefix for the classes that are renamed with `.esclass` extension.